imap_processing/hi/l1a/hi_l1a.py

```
@@ -26,27 +27,31 @@ def hi_l1a(packet_file_path: str):
     """
     unpacked_data = decom_hi.decom_packets(packet_file_path)
 
-    # group data by appId
+    # group data by apid
     grouped_data = group_by_apid(unpacked_data)
 
     # Process science to l1a.
     processed_data = []
     for apid in grouped_data.keys():
         if apid == HIAPID.H45_SCI_CNT:
-            # TODO: Add processing for science count data
-            continue
+            logger.info(
+                "Processing histogram data for [%s] packets", HIAPID.H45_SCI_CNT.name
+            )
+            data = hist_create_dataset(grouped_data[apid])
         elif apid == HIAPID.H45_SCI_DE:
             logger.info(
-                "Processing direct event data for [%s] packets", HIAPID.H45_SCI_CNT.name
+                "Processing direct event data for [%s] packets", HIAPID.H45_SCI_DE.name
             )
 
             data = science_direct_event(grouped_data[apid])
-            processed_data.append(data)
         elif apid == HIAPID.H45_APP_NHK:
             logger.info(
                 "Processing housekeeping data for [%s] packets", HIAPID.H45_APP_NHK.name
             )
-            data = create_dataset(grouped_data[apid])
-            processed_data.append(data)
+            data = hk_create_dataset(grouped_data[apid])
+        else:
+            logger.warning(f"Encountered unexpected APID [{apid}]")
+            continue
```

Is this OK to have happen, or should we be raising RunTimeError when this occurs to attempt to abort?

---

Good question. I think we should log it because if we have other good apid that we could process, it would be good to process those. But good point about how to raise this error without aborting all.

---

Valid point.

---

I'm not sure how to resolve this. Getting an unexpected APID would indicate a bug in something upstream that didn't filter APIDs correctly or called the wrong processing function for a given L0 file. I don't think that we should plan for this to happen regularly so it seems like raising a RunTimeError would be appropriate in order to fail loudly.

---

I share your concern about a logger warning not being loud enough and possibly going undetected which is why I've been stressing the raise instead of warn. But, I don't have a major preference on that and it is likely not a big deal. You can choose what you feel is appropriate here and we can change things later if we need to, I don't think anyone is blocking over this question.